kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/BaseOperation.java

```diff
@@ -302,6 +302,36 @@ public T apply(T resource) {
     }
   }
 
+
+  @Override
+  public D createOrReplaceWithNew() throws KubernetesClientException {
+    final Function<T, T> visitor = new Function<T, T>() {
+      @Override
+      public T apply(T resource) {
+        try {
+          return createOrReplace(resource);
+        } catch (Exception e) {
+          throw KubernetesClientException.launderThrowable(e);
+        }
+      }
+    };
+
+    try {
+      return getDoneableType().getDeclaredConstructor(Function.class).newInstance(visitor);
+    } catch (InvocationTargetException | NoSuchMethodException | IllegalAccessException | InstantiationException e) {
+      throw KubernetesClientException.launderThrowable(e);
+    }
+  }
+
+  @Override
+  public T createOrReplace(T item) {
+    if (get() == null) {
```

Comment:
@iocanel question regarding `resource().createOrReplace(resourceObj);`

the `get()` will return a `null` value if and only if none of the namespaces already contain the `Resource`?
which means that the `create` workflow will only be invoked during the first call to create a `Resource` in any `namespace`. All subsequent calls to create a similar `Resource` in a different `namespace` will invoke the `replace` workflow and subsequently fail because the `Resource` doesn't exist?

example scenario: ( will succeed only if there are no ResourceQuotas in any of the namespaces. )
`client.ResourceQuotas().createOrReplace(resourceQuota); `

this attempts a replace as well: (the get() returns the loaded object from context and fails with PUT on non-existent resource)
`client.resourceQuotas().inNamespace(namespace).load(resourceQuotaInputStream).createOrReplace(resourceQuota);`

is there a way to support the following operation?
`client.ResourceQuotas().inNamespace(namespace).createOrReplace(resourceQuota);`